panther_analysis_tool/policy.py

```diff
@@ -0,0 +1,47 @@
+"""
+Panther Analysis Tool is a command line interface for writing,
+testing, and packaging policies/rules.
+Copyright (C) 2020 Panther Labs Inc
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from typing import List
+
+from panther_analysis_tool.enriched_event import PantherEvent
+from panther_analysis_tool.rule import Detection
+
+TYPE_POLICY = "POLICY"
+
+ERROR_TYPE_POLICY = "POLICY_ERROR"
+
+
+class Policy(Detection):
+    """Panther policy metadata and imported module."""
+
+    # default detection types for policies
+    default_detection_type = TYPE_POLICY
+
+    # policies have a rule method
+    matcher_function_name = "policy"
+
+    # a policy should trigger an alert on False return value
+    matcher_alert_value = False
+
+    # suppressions for the policy
+    suppressions: List[str] = []
+
+    def matcher_function(self, event: PantherEvent) -> bool:
```

Comment:
Hmm does it make sense to have a `PantherEvent` type describing a resource?

In the current world, I think the answer is no.

However, since the purpose of this is in support of unifying the policy & rule pipelines to a single experience, I think it makes a certain amount of sense. My only concern really is that if we add some functionality to  `PantherEvent`s that really isn't applicable to resources then that could be weird & unintuitive.

Also, I don't have an idea of what a more generic name would be. So I think `PantherEvent` is fine. Just wanted to raise the concern that we're blurring two different concepts into one here.

Reply:
Yea, I was thinking converting to `PantherEvent` was useful because it makes the json immutable - but I don't know if that is necessarily needed for resources because we don't store the output in the datalake like we do matched events. I can change this to a standard `dict()` - since we also don't currently (/ won't?) use data models in policies, which is the other reason to use this type.

Reply:
One extra benefit is the case-insensitive lookups, which just make authoring a policy bit easier. I'm guessing though that policies are written more rarely and the resource structure is very specific, so this might not be a real gain. 

Regarding the immutability feature, it seems that the engine [depends on some of the resource fields](https://github.com/panther-labs/panther-enterprise/blob/25a70fbf323b23c7d6fcfaaf633071767d7b7246/internal/compliance/policy_engine/src/policy.py#L285) to be accurate in the result, not sure how critical that is though.